.github/workflows/scripts/e2e-maven-push.sh

```diff
@@ -0,0 +1,152 @@
+# because you cannot overwrite a version in maven. Instead we commit to main,
+# set the tag, reset $branch and push both main and $branch.
+echo "GITHUB_REPOSITORY: ${GITHUB_REPOSITORY}"
+gh repo clone "${GITHUB_REPOSITORY}" -- -b maven-e2e-temp
+repo_name=$(echo "$GITHUB_REPOSITORY" | cut -d '/' -f2)
+cd ./"$repo_name"
+
+git config --global user.name github-actions
+git config --global user.email github-actions@github.com
+
+# Set the remote url to authenticate using the token.
+git remote set-url origin "https://github-actions:${GH_TOKEN}@github.com/${GITHUB_REPOSITORY}.git"
+
+package_dir="${PACKAGE_DIR}" # specified in the e2e test yaml
+
+cd "${package_dir}"
+
+# Get the new version
+artifact_tag=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+
+# version_major prints the major version number.
+# Expects a string like '1.19.7'
+# version_major returns "1" if the input is '1.19.7'
+version_major() {
+    VER=$(echo $1 | cut -d '.' -f1)
+    echo "$VER"
+}
+
+# version_minor prints the minor version number.
+# Expects a string like '1.19.7'.
+# version_minor returns "19" if the input is '1.19.7'
+version_minor() {
+    VER=$(echo $1 | cut -d '.' -f2)
+    echo "$VER"
+}
+
+# version_patch prints the patch version number.
+# Expects a string like '1.19.7-SNAPSHOT.jar'
+# version_patch returns "7" if the input is '1.19.7'
+version_patch() {
+    VER=$(echo $1 | cut -d '.' -f3)
+    echo "$VER"
+}
+
+# Bumps the version
+new_version() {
+    current_tag=$1
+    release_major=$(version_major "$current_tag")
+    release_minor=$(version_minor "$current_tag")
+    release_patch=$(version_patch "$current_tag")
+
+    # These if-statements are sorted by likelihood
+    if [[ $release_patch != "99" ]]; then
+        # Only need to bump the patch
+        release_patch=$((release_patch+1))
+    elif [[ $release_patch = "99" && $release_minor != "99" ]]; then
+        # Need to bump minor
+        release_minor=$(($release_minor+1))
+        release_patch="0"
+    elif [[ $release_patch = "99" && $release_minor = "99" ]]; then
+        # Need to bump major
+        release_major=$(($release_major+1))
+        release_minor="0"
+        release_patch="0"
+    fi
+    echo $release_major.$release_minor.$release_patch
+}
+
+next_tag=$(new_version $artifact_tag)
+
+# Output the artifact name
+echo "artifact-version=${artifact_tag}" >> $GITHUB_OUTPUT
+
+tag=$(mvn versions:set -DnewVersion=$next_tag)
+cd -
+
+# Commit the new version.
+git commit -m "${GITHUB_WORKFLOW}" "${package_dir}/pom.xml" "${package_dir}/pom.xml"
+
+# If this is an e2e test for a tag, then tag the commit and push it.
+this_event=$(e2e_this_event)
+echo "this_event: ${this_event}"
+if [ "${this_event}" == "tag" ] || [ "${this_event}" == "create" ]; then
+    git tag "${tag}"
+fi
+
+git remote -v
+git branch
+pwd
+if [ "${branch}" != "main" ]; then
+    # Reset branch1 and push the new version.
+    # git branch -D "$branch"
+    git checkout -b "$branch"
```

Comment:
nit: might as well be consistent
```suggestion
    git checkout -b "${branch}"
```